components/views/auth/ResetPassword.tsx

```diff
@@ -1,38 +1,116 @@
+import {
+    showTechnicalDifficultiesBanner,
+    showBanner,
+    hideBanner
+} from '../../../lib/redux/slices/banner'
 import { FormRow, PasswordField, SubmitButton } from '../../form'
 import FormValidations from '../../../utilities/validations/forms'
 import styles from './AuthPages.module.scss'
-
+import { useDispatch, useSelector } from 'react-redux'
+import { AppDispatch } from '../../../lib/redux/store'
+import { setLoading, setProfile } from '../../../lib/redux/slices/user'
+import { JWT_SECRET } from '../../../utilities/constants'
+import { RESET_PASSWORD } from '../../../lib/gql/mutations/users'
 const INITIAL_STATE = {
     newPassword: { value: '', error: '' },
     confirmPassword: { value: '', error: '' }
 } as FormProps
 
 const VALIDATIONS = {
     newPassword: FormValidations.validPassword,
-    confirmPassword: FormValidations.validConfirmedPassword
+    confirmPassword: FormValidations.validPassword
 }
 
 const ResetPassword = () => {
-    const { form, handleChange, isFormValid } = useForm(
+    const dispatch = useDispatch<AppDispatch>()
+    const router = useRouter()
+    const user = useSelector((state: any) => state.user)
+    const [disableSubmit, setDisableSubmit] = useState<boolean>(true)
+    const { form, handleChange, isFormValid, handleReset } = useForm(
         INITIAL_STATE,
         VALIDATIONS
     )
     const { newPassword, confirmPassword } = form
 
+    const [resetPassword] = useMutation(RESET_PASSWORD, {
+        onCompleted({ register: data }) {
+            if (data.__typename === 'Errors') {
+                dispatch(
+                    showBanner({
+                        message: data.message,
+                        type: data.__typename
+                    })
+                )
+            } else {
+                const profile = { ...data }
+                const token = data.token
+                localStorage.setItem(JWT_SECRET, token)
+
+                delete profile.__typename
+                delete profile.successType
+                delete profile.token
+                dispatch(setProfile(profile))
+
+                dispatch(
+                    showBanner({
+                        message: 'Your password has sucessfully been changed.',
+                        type: data.__typename
+                    })
+                )
+                handleReset()
+            }
+            dispatch(setLoading(false))
+            router.push('/')
+        },
+        onError(err: any) {
+            dispatch(setLoading(false))
+            dispatch(showTechnicalDifficultiesBanner())
+        },
+        variables: {
+            newPassword: newPassword.value,
+            confirmPassword: confirmPassword.value
+        }
+    })
+
     const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
         event.preventDefault()
+        dispatch(setLoading(true))
+        resetPassword()
     }
 
+    useEffect(() => {
+        let isValid = true
```

Comment:
I got ride of this line in my comment for a reason 🤣 it's not needed. This will only ever be `true` or `false` based on one check, not a series of checks, so having a `let` is not needed. 

`const isValid = !passwordsMatch && isFormValid ` is the exact same thing as your `let` and second definition here. Small steps to cleaner and more efficient code.

Also, there's no reason to do the 2nd `if (isFormValid)` at the bottom. It can all be checked in the one line above, then referenced together, i gave you the code in the comment 🤣

Reply:
fixed. swapped `const isValid = !passwordsMatch && isFormValid` to `const isValid = passwordsMatch && isFormValid` so the logic works